### DISCUSSION TITLE
RSAT2 data processing w/stripmapStack

### DISCUSSION THREAD

**Discussion by 10go-L** [isce-framework/isce2#398] [Q&A]
Hello, after googling and a lot of trials and errors, I am hitting a wall. I am using the licensed version of ISCE-2.2.0 from https://winsar.unavco.org/software/isce. I also built and installed the ISCE-2.0.0 version without success.

My main goal is to use stripmapStack to process this data and at the moment I have the following errors:
**ISCE-2.2.0**

- stripmapApp.py
`stripmapApp.py --steps int_130327_140322.xml`
![image](https://user-images.githubusercontent.com/91211678/143057503-ab64f83c-6727-44fd-9bab-e7e48cf5a715.png)
- unpackFrame_RSAT2.py
`unpackFrame_RSAT2.py -i ../rsat2_data/130327/ -o SLC/20130327`
![image](https://user-images.githubusercontent.com/91211678/143057665-02eb338e-46af-4f6a-b596-d2b7c2e9f41a.png)

I got the same error, so I tried to change the versions of gfortran, gcc, and g++ to 4.6, 5.5, and 7.5 without success. So the active one on those screenshots is version 4.6.

**ISCE-2.0.0**
Using isce-2.0.0, I was able to see more progress in the processing, but in the end, I got an error as well:
`insarApp.py --steps insar_130327_140322.xml`
![image](https://user-images.githubusercontent.com/91211678/143058590-3c3a89d2-459c-410a-93f5-00f5ebb738cd.png)

As I mentioned before, my main goal is to process this data with stripmapStack.py. Any suggestion would be helpful.

Thanks!

Edit:
Forgot to mention, when using the non-licensed version of isce installed via conda I have the following errors:
**ISCE-2.5.3**
`unpackFrame_RSAT2.py -i ../rsat2_data/130327/ -o SLC/20130327`
![image](https://user-images.githubusercontent.com/91211678/143060702-606e1cf6-1858-4fdf-b6f3-3aa73fbd949d.png)

**Comment by piyushrpt** [comment]
Unless you have access to RSAT2 precise orbits, this is not going to be easy to deal with. If you have access to precise orbits, you should be able to unpack the data successfully. The orbits in XML file are often of insufficient duration and we had a hacky work around to extend it (related to errors above).

**Reply by 10go-L** [reply]
Hello Piyush,

Thanks for your quick reply.  I am not sure, but I don't think I have RSAT2 precise orbits; this is the data I'm working with:
![image](https://user-images.githubusercontent.com/91211678/143076023-bef88142-3d29-4de5-95f8-e603a544fb28.png)

Do you have any idea if it is possible to go further with the process?

Related to this error in particular:
![image](https://user-images.githubusercontent.com/91211678/143076410-ed1665dd-2ba4-4e64-9b5c-739f4f19bf16.png)
I thought it was related to a different package of gfortran or gcc or g++. Have you experienced it before?

**Comment by piyushrpt** [comment]
They are separate files that are distributed separately and not included with the imagery. If you don't have access to those, it is going to be hard to make RSAT2 work like other sensors. The orbits in XML files are not of great quality and you actually have to use a different orbit interpolation method etc. Easiest would be to work with the precise orbits and pass the folder with the orbits as an input to the reader object using `orbitDirectory` property like ERS/ENV/S1 during unpacking.

**Reply by Sugeradi985** [reply]
Hi Piyush,
So where can we get the precise orbits? Are they available to data agents？
Thanks.

**Reply by piyushrpt** [reply]
You may have to ask your provider. How much of information is openly available to everyone or to select users only is never very clear for these traditional commercial providers.
